polyA/fill_align_matrix.py

```diff
@@ -79,6 +84,12 @@ def fill_align_matrix(
         gap_init = gap_inits[i]
         gap_ext = gap_exts[i]
 
+        char_complexity_adjustments: Dict[
+            str, int
+        ] = calculate_complexity_adjusted_score(
```

Comment:
Strictly speaking you don't need the type annotation on the variable as long as the function as a return type annotation. So I would move the type annotation to `calculate_complexity_adjusted_score`. For example:

```
def foo() -> int:
  pass
```

defines a function that returns an `int`.